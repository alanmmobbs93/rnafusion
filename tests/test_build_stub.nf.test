nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    profile "test_build"
    options "-stub" // TODO remove once the pipeline is fixed
    tag "pipeline"
    tag "pipeline_rnafusion"
    tag "test_build_stub"

    test("stub test_build with fastp_trim") {

        when {
            params {
                outdir          = "$outputDir"
                fastp_trim      = true
                cosmic_username = "user" // placeholder string to avoid null input val in process
                cosmic_passwd   = "password" // placeholder string to avoid null input val in process
                all             = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("stub test_build no fastp_trim") {

        when {
            params {
                outdir          = "$outputDir"
                fastp_trim      = true
                cosmic_username = "user" // placeholder string to avoid null input val in process
                cosmic_passwd   = "password" // placeholder string to avoid null input val in process
                all             = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
