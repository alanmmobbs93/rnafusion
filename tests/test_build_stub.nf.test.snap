{
    "stub test_build": {
        "content": [
            14,
            [
                "bedops",
                "bedops/Homo_sapiens.GRCh38.102_rrna_intervals.gtf.bed",
                "gatk4",
                "gatk4/Homo_sapiens.GRCh38.102.dna.primary_assembly.dict",
                "gffread",
                "gffread/Homo_sapiens.GRCh38.102.gff3",
                "pipeline_info",
                "references",
                "references/arriba",
                "references/arriba/blacklist_hg38_GRCh38_v2.4.0.tsv.gz",
                "references/arriba/cytobands_hg38_GRCh38_v2.4.0.tsv",
                "references/arriba/known_fusions_hg38_GRCh38_v2.4.0.tsv.gz",
                "references/arriba/protein_domains_hg38_GRCh38_v2.4.0.gff3",
                "references/ensembl",
                "references/ensembl/Homo_sapiens.GRCh38.102.chr.gtf",
                "references/ensembl/Homo_sapiens.GRCh38.102.dna.primary_assembly.fa",
                "references/ensembl/Homo_sapiens.GRCh38.102.dna.primary_assembly.fa.fai",
                "references/ensembl/Homo_sapiens.GRCh38.102.genepred",
                "references/ensembl/Homo_sapiens.GRCh38.102.gtf",
                "references/ensembl/Homo_sapiens.GRCh38.102.refflat",
                "references/ensembl/Homo_sapiens.GRCh38.102_rrna_intervals.gtf.interval_list",
                "references/fusion_report_db",
                "references/fusion_report_db/cosmic.db",
                "references/fusion_report_db/fusion_report.log",
                "references/fusion_report_db/fusiongdb2.db",
                "references/fusion_report_db/mitelman.db",
                "references/fusioncatcher",
                "references/fusioncatcher/human_v102",
                "references/hgnc",
                "references/hgnc/HGNC-DB-timestamp.txt",
                "references/hgnc/hgnc_complete_set.txt",
                "references/star",
                "references/star/Genome",
                "references/star/Log.out",
                "references/star/SA",
                "references/star/SAindex",
                "references/star/chrLength.txt",
                "references/star/chrName.txt",
                "references/star/chrNameLength.txt",
                "references/star/chrStart.txt",
                "references/star/exonGeTrInfo.tab",
                "references/star/exonInfo.tab",
                "references/star/geneInfo.tab",
                "references/star/genomeParameters.txt",
                "references/star/sjdbInfo.txt",
                "references/star/sjdbList.fromGTF.out.tab",
                "references/star/sjdbList.out.tab",
                "references/star/transcriptInfo.tab",
                "references/starfusion",
                "references/starfusion/ctat_genome_lib_build_dir",
                "references/starfusion/ref_annot.cdna.fa",
                "rrnatranscripts",
                "rrnatranscripts/Homo_sapiens.GRCh38.102_rrna_intervals.gtf"
            ],
            [
                "Homo_sapiens.GRCh38.102_rrna_intervals.gtf.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.dna.primary_assembly.dict:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.gff3:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cytobands_hg38_GRCh38_v2.4.0.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "protein_domains_hg38_GRCh38_v2.4.0.gff3:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.chr.gtf:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.dna.primary_assembly.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.dna.primary_assembly.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.genepred:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.gtf:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102.refflat:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102_rrna_intervals.gtf.interval_list:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cosmic.db:md5,d41d8cd98f00b204e9800998ecf8427e",
                "fusion_report.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                "fusiongdb2.db:md5,d41d8cd98f00b204e9800998ecf8427e",
                "mitelman.db:md5,d41d8cd98f00b204e9800998ecf8427e",
                "HGNC-DB-timestamp.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "hgnc_complete_set.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Genome:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Log.out:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SA:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SAindex:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrLength.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrName.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrNameLength.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrStart.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "exonGeTrInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "exonInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "geneInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "genomeParameters.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "sjdbInfo.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "sjdbList.fromGTF.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "sjdbList.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "transcriptInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.cdna.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Homo_sapiens.GRCh38.102_rrna_intervals.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-29T17:29:03.231447837"
    }
}